cmd/juju/status/status.go

```diff
@@ -411,14 +410,28 @@ func (c *statusCommand) Run(ctx *cmd.Context) error {
 	defer c.close()
 
 	if c.watch != 0 {
-		jujuStatusArgs := c.statusCommandForViddy(os.Args)
-
-		viddyArgs := append([]string{"--no-title", "--interval", c.watch.String()}, jujuStatusArgs...)
-
-		// Define tview styles and launch preconfiged Viddy watcher
-		app := viddy.NewPreconfigedViddy(viddyArgs)
-		if err := app.Run(); err != nil {
-			return errors.Annotate(err, "unable to run Viddy (watcher for status command)")
+		// Collect all args except the watch flag
+		jujuStatusArgs := c.statusCommandForWatch(os.Args)
+
+		// Check what watch shell command is available. Using the following order:
+		// 1. `viddy`
+		// 2. `watch`
+		// 3. send error to output
+		var watchArgs []string
+		var watchableJujuCmd *exec.Cmd
+		if _, err := exec.LookPath("viddy"); err == nil {
+			watchArgs = append([]string{"--no-title", "--interval", c.watch.String()}, jujuStatusArgs...)
+			watchableJujuCmd = exec.Command("viddy", watchArgs...)
+		} else if _, err = exec.LookPath("watch"); err == nil {
+			watchArgs = append([]string{"-n", strconv.Itoa(int(c.watch.Seconds()))}, jujuStatusArgs...)
```

Comment:
From inside the strictly confined juju snap, this will probably error in strange ways. Since watch is findable in the PATH but not in the juju snap.

Reply:
Agreed.

I reorganized the logic a bit. And add `viddy` as a `part` of juju snap.

I suppose it's better than a "dependency via custom fork".

Reply:
Updated code, and added QA of how it works with juju strictly confined snap.